### DISCUSSION TITLE
Local Notifications and PWA's

### DISCUSSION THREAD

**Discussion by mark-divitt** [ionic-team/capacitor#3220] [Q&A]
Hi Folks,
             I am trying to get Local Notifications working with my PWA.   Works great on my (Desktop) Chrome browser, but fails on Chrome on a Samsung S9.  Here is the error in the console : 
  
```
capacitor.js:2 Uncaught TypeError: Failed to construct 'Notification': Illegal constructor. Use ServiceWorkerRegistration.showNotification() instead.
    at t.buildNotification (capacitor.js:2)
    at capacitor.js:2
    at Array.forEach (<anonymous>)
    at t.sendPending (capacitor.js:2)
    at capacitor.js:2
```
I am doing a   ```const checkperm = await LocalNotifications.requestPermission();``` before I call 
```
const notifs = await LocalNotifications.schedule({
      notifications: [...
```
I have notifications turned on (specifically) for Chrome on the Samsung S9 as well as globally.  So there is something about the S9 (either browser/OS or config) that is different.

Any tips or places to look would be greatly appreciated.

Thanks,
Mark

**Comment by mark-divitt** [comment]
Bump.

**Comment by mark-divitt** [comment]
Any clues would be appreciated...

**Comment by mark-divitt** [comment]
Bueller?  Bueller?

**Comment by mark-divitt** [comment]
Bump...

**Comment by mhartington** [comment]
@mark-divitt You do not need to keep "bumping" the post. We see it.

**Comment by mark-divitt** [comment]
Thank you.  My intention is not to be difficult, just stuck and need some
ideas to get me unstuck...

**Comment by ffranchina** [comment]
Hello everyone! Sorry for resurrecting this very old discussion but I think we should seriously address such issue.

So far the `LocalNotifications` API, if used as web native API, can be successfully used just on the desktop and on less spread browsers (like Firefox for Android). Currently the API cannot be used inside installable PWA that rely on Chrome for Android, for instance and I guess this is an unwanted behavior.

I researched on the matter ([here](https://stackoverflow.com/questions/29774836/failed-to-construct-notification-illegal-constructor) I got the key insights) and I discovered that the issue is given by the progresses made by the [Web Notification API itself](https://notifications.spec.whatwg.org/#api):

> A [non-persistent notification](https://notifications.spec.whatwg.org/#non-persistent-notification) is represented by one [Notification](https://notifications.spec.whatwg.org/#notification) object and can be created through [Notification](https://notifications.spec.whatwg.org/#notification)'s [constructor](https://notifications.spec.whatwg.org/#dom-notification-notification).
>A [persistent notification](https://notifications.spec.whatwg.org/#persistent-notification) is represented by zero or more [Notification](https://notifications.spec.whatwg.org/#notification) objects and can be created through the [showNotification()](https://notifications.spec.whatwg.org/#dom-serviceworkerregistration-shownotification) method.

At the moment the `LocalNotifications` API of Capacitor doesn't take the latter case into account and fails in most of the functions exposed.
Hacking a bit with the console, though, I managed to have working notifications on my PWA on Chrome ([here](https://developer.mozilla.org/en-US/docs/Web/API/ServiceWorkerRegistration/showNotification#examples) the code for launching the notification), so it is technically possible.

My question for you is: do we want to change the plugin to accommodate such needs or you think they are out of the scope and would be better to write a different plugin?

Thanks for the attentions, I will look forward for your comments.
